packages/framework-core/src/services/graphql/graphql-generator.ts

```diff
@@ -90,10 +104,19 @@ export class GraphQLGenerator {
   ): GraphQLFieldResolver<unknown, GraphQLResolverContext, ReadModelRequestArgs<ReadModelInterface>> {
     return (parent, args, context, info) => {
       let isPaginated = false
+      const fields: ProjectionFor<unknown> = this.getFields(info) as ProjectionFor<unknown>
+      let select: ProjectionFor<unknown> | undefined = fields.length > 0 ? fields : undefined
       if (info?.fieldName === `List${pluralize(readModelClass.name)}`) {
         isPaginated = true
+        if (select) {
+          select = select
+            .map((field: string) => {
+              return field.split('.').slice(1).join('.')
+            })
+            .filter((str: string) => str.trim().length > 0) as ProjectionFor<unknown>
+        }
```

Comment:
I'm still unsure why you are doing this map. We are removing the first component of every projection path. What for? 
And if we are always removing, why returning it in the first place? We can make the `this.getFields` method not to return it.

Reply:
@alvaroloes this `map` takes care of the `List<Read Model name>` GraphQL queries (paginated queries), which will wrap the fields inside an `items` property. For example:

```graphql
query ListMyReadModels {
  items {
    foo
    bar
  }
}
```

Calling `getFields` on this query will return `['items[].foo', 'items[].bar']`, but our read model, in this case, `MyReadModel`, doesn't have an `items` property when stored on the database, it has `foo` and `bar` properties, so the actual fields we want to select are these two. Every time one of these types of queries comes, it will have an `items[]` prepended. This `map` cleans the `select`.

Reply:
@MarcAstr0 Ahhh! Got it, thanks. Do you mind writing a comment saying that please 🙏 ? Something like "Removing `items[].` from the selection set". That way future us or other devs will have an easier time understanding it. Thanks a lot!